components/transactions/TransactionRejectButton.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { FormattedMessage } from 'react-intl';
+
+import { getErrorFromGraphqlException } from '../../lib/errors/';
+import { GraphQLContext } from '../../lib/graphql/context';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+
+import { Box, Flex } from '../Grid';
+import MessageBox from '../MessageBox';
+import StyledButton from '../StyledButton';
+
+/* eslint-disable graphql/template-strings */
+export const rejectTransactionMutation = gqlV2/* GraphQL */ `
+  mutation RejectTransaction($transaction: TransactionReferenceInput!) {
+    rejectTransaction(transaction: $transaction) {
+      id
```

Comment:
If we fetch the fields that we want to update here, we won't even have to refetch the data because Apollo will be able to update the cache. We should fetch the order info (for the status), and the refund transaction.